### PR TITLE
Minor enhancements

### PR DESCRIPTION
**Pull request by davidmisiak**
This PR fixes https://github.com/vacuumlabs/cardano-hw-cli/issues/120, https://github.com/vacuumlabs/cardano-hw-cli/issues/126, https://github.com/vacuumlabs/cardano-hw-cli/issues/131, https://github.com/vacuumlabs/cardano-hw-cli/issues/132, https://github.com/vacuumlabs/cardano-hw-cli/issues/134 and allows non-Plutus signing modes for txs with required signers. Most of the changes are rather small and simple, therefore only a single PR instead of six, but I recommend reviewing commit-by-commit.

### CODE REVIEW

**Comment by gitmachtl** [comment]
#132 so the restriction has been lifted and now the payment key from a hw-wallet can also be used for any kind of pool-registration/pool-update transactions?

**Comment by davidmisiak** [comment]
> https://github.com/vacuumlabs/cardano-hw-cli/issues/132 so the restriction has been lifted and now the payment key from a hw-wallet can also be used for any kind of pool-registration/pool-update transactions?

No, pool registration still cannot be signed with a payment key as a pool owner (i.e. by a pyament and a staking key at the same time). Signing only with a payment key as the pool operator was allowed by the Ledger app (if I'm not mistaken), it just hasn't been supported by hw-cli.

File: test/integration/ledger/node/tx.js [hunk 1]
```diff
@@ -64,40 +64,40 @@ const transactions = {
...
+    signedTxCborHex: '83a50081825820bc8bf52ea894fb8e442fe3eea628be87d0c9a37baef185b70eb00a5c8a849d3b0001818258390114c16d7f43243bd81478e68b9db53a8528fd4fb1078d58d54a7f11241d227aefa4b773149170885aadba30aab3127cc611ddbc4999def61c1a00311cba021a0002c431031a00ac30b105a1581de11d227aefa4b773149170885aadba30aab3127cc611ddbc4999def61c1a000ded3aa1008282582066610efd336e1137c525937b76511fbcf2a0e6bcf0d340a67bcb39bc870d85e85840eeb76938fd9676a34ba0710dc27810d087507b3a75ac6f67543b0b22405c927ad9f575a258aa7ed1dd1bbf3d24596315ffba8d630e0a1ea8d105826b865b3808825820cd2b047d1a803eee059769cffb3dfd0a4b9327e55bc78aa962d9bd4f720db0b25840501a09efd212efd741574e63e9ff6c701746cac68ddcba3af5ef655ff1e724399adef1eb258ffdb34fd09d7b91c4b2f612bfba083b2debaa87ed93fcf4bc1f08f6',
```
Comment (janmazak) [inline test/integration/ledger/node/tx.js:67]: Why is this changed? Just because the order of witnesses has changed?

File: src/types.ts [hunk 2]
```diff
@@ -66,6 +66,12 @@ export type TxFileData = {
 
 export type Address = string
 
+export enum DerivationType {
```
Comment (janmazak) [inline src/types.ts:69]: Perhaps a comment that it only applies to Trezor, with a link to a description of the derivation types.

File: src/crypto-providers/util.ts [hunk 3]
```diff
@@ -121,6 +121,8 @@ const pathEquals = (path1: BIP32Path, path2: BIP32Path) => (
   path1.length === path2.length && path1.every((element, i) => element === path2[i])
 )
 
+const isByronPath = (path: BIP32Path) => classifyPath(path) === PathTypes.PATH_WALLET_SPENDING_KEY_BYRON
```
Comment (janmazak) [inline src/crypto-providers/util.ts:124]: I don't like this much, the name is deceptive --- the function returns false for a Byron account path. Do we need a "global" function like this?

**Review by janmazak — APPROVED** [review]
Maybe a little refactor of isByron, but LGTM. I've not run any tests, nor searched all the source code for missed items to rename etc.

File: test/integration/ledger/node/tx.js (hunk 1 above)
Comment (davidmisiak) [reply test/integration/ledger/node/tx.js:67]: Yes

File: src/crypto-providers/util.ts (hunk 3 above)
Comment (davidmisiak) [reply src/crypto-providers/util.ts:124]: It's used on several places in two files, but the condition is simple, so maybe we could inline it.

File: src/types.ts (hunk 2 above)
Comment (davidmisiak) [reply src/types.ts:69]: Added

File: src/crypto-providers/util.ts (hunk 3 above)
Comment (davidmisiak) [reply src/crypto-providers/util.ts:124]: Inlined

**Comment by gitmachtl** [comment]
For release notes 1.12.0:

* keeping original auxiliary data hash on transaction transform when the tx doesn't contain auxiliary data ([cardano-hw-interop-lib](https://github.com/vacuumlabs/cardano-hw-interop-lib/releases/tag/v1.1.1))

that line is a bit misleading. the auxdata hash is kept if no auxiliary data is in the tx, and its corrected if the auxdata was reordered in canonical order. or is the auxdata not reordered anymore?

**Comment by davidmisiak** [comment]
> the auxdata hash is kept if no auxiliary data is in the tx, and its corrected if the auxdata was reordered in canonical order

Yes, that's the case. I can try to clarify that in the release notes.
